src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PrivateDataRetrievalService.java

```diff
@@ -63,19 +72,26 @@ private ManagingOfficerListDataApi getManagingOfficerData(String companyNumber)
 
   private OverseasEntityDataApi getOverseasEntityData(String companyNumber)
       throws ServiceException {
+    var logMap = new HashMap<String, Object>();
     try {
       OverseasEntityDataApi overseasEntityDataApi = apiClientService
           .getInternalApiClient()
           .privateOverseasEntityDataHandler()
           .getOverseasEntityData(OVERSEAS_ENTITY_URI_SECTION + companyNumber + "/entity-data")
           .execute()
           .getData();
-      var logMap = new HashMap<String, Object>();
+
 
       logMap.put(COMPANY_NUMBER, companyNumber);
       ApiLogger.info("Retrieving overseas entity data for company number ",  logMap);
 
       return overseasEntityDataApi;
+    } catch (ApiErrorResponseException e) {
+      if (e.getStatusCode() == HttpServletResponse.SC_NOT_FOUND) {
+        ApiLogger.info(String.format("No Managing Officers data found for Company Number %s", companyNumber), logMap);
+        return null;
+      }
+      throw new ServiceException(e.getStatusMessage(), e);
```

Comment:
For overseas entity 404 is an error and should not be masked. Remove new catch for getCompanyProfile